File: server-practice/message-server/message-application/application-post/src/main/kotlin/dev/lkdcode/application/ports/input/command/CreatePostUsecase.kt
package dev.lkdcode.application.ports.input.command

import dev.lkdcode.domain.entity.Post
import dev.lkdcode.domain.value.Content
import dev.lkdcode.domain.value.Title

interface CreatePostUsecase {
    fun create(dto: CreatePostDTO): Post
}

data class CreatePostDTO(
    val title: String,
    val content: String,
    val authorId: Long,
) {
    val titleValue get() = Title(title)
    val contentValue get() = Content(content)
}

File: server-practice/py-server/py_application/post/ports/input/create_post_usecase.py
from abc import ABC, abstractmethod
from dataclasses import dataclass
from py_domain.post.entity.post import Post

@dataclass
class CreatePostDTO:
    title: str
    content: str

class CreatePostUsecase(ABC):
    @abstractmethod
    def create(self, dto: CreatePostDTO) -> Post:
        pass

